pkg/controller/shared_state.go

```diff
@@ -1536,18 +1537,24 @@ func (c *SharedState) createAzVolume(ctx context.Context, desiredAzVolume *azdis
 
 	var err error
 	var azVolume *azdiskv1beta2.AzVolume
+	originalAzVolume := azVolume.DeepCopy()
```

Comment:
`azVolume` is `nil` so this does nothing but declare a variable. You can simple do so directly.

Reply:
We should need to copy the `AzVolume` object though. What's the reason for this change?

Reply:
No special reason. I was dumb lol. I changed it back to `var originalAzVolume *azdiskv1beta2.AzVolume`. `DeepCopy()` should be done later.